policy/file/source.go

```diff
@@ -245,13 +265,13 @@ func (s *Source) handleDir() ([]policy.PolicyID, error) {
 			continue
 		}
 
-		scalingPolicy.ApplyDefaults(s.config)
-
-		// We have had to decode the file, so store the information. This makes
-		// the MonitorPolicy function simpler as we have an easy mapping of the
+		// Store the file>id mapping if it doesn't exist. This makes the
+		// MonitorPolicy function simpler as we have an easy mapping of the
 		// policyID to the file it came from.
 		s.policyMapLock.Lock()
-		s.policyMap[policyID] = &filePolicy{file: file, policy: &scalingPolicy}
+		if _, ok := s.policyMap[policyID]; !ok {
+			s.policyMap[policyID] = &filePolicy{file: file}
+		}
```

Comment:
why the `ok` check instead of just overwriting what's there? is there concurrent code which expects the object at `s.policyMap[policyID]` to remain unchanged (i didn't see any)?

Reply:
This part caused by about an hour of head scratching. 

This function gets called during the initial load and then on reloads of the monitor IDs routine. When we are asked to reload, if we overwrite what is stored, when we subsequently trigger reload of the individual policy monitor we can never tell whether the newly read policy differs from the stored policy.

Reply:
:100: 

should probably add a comment to that effect 😄